synapse/lib/hiveauth.py

```diff
@@ -3,6 +3,8 @@
 
 from typing import Union
 
+import regex
```

Comment:
why regex and not `re` ?

Reply:
Because `re` lacks a lot of functionality compared to `regex`.

Reply:
I also think I saw somewhere that `regex` is a lot faster than `re`. Probably not a major consideration with the way we're using it here but still a valid point.

Reply:
Got my wires crossed!